GetStream/stream-chat-android
# Add channel capabilities

closes #2922 

### 🎯 Goal

Add channel capabilities.

### 🛠 Implementation details
1. Added `DownstreamChannelDto::own_capabilities: List<String>`
2. Added `Channel::ownCapabilities: Set<String>`
3. Added `ChannelCapabilities` object with predefined values
4. Added enriching `Channel` with capabilities when storing channel from the event

### 🧪 Testing

1. Check if channels contain capabilities.
2. Make sure capabilities don't disappear after receiving events (e.g. `ChannelUpdated`).
3. Check if all capabilities from [docs](https://getstream.io/chat/docs/javascript/channel_capabilities/?language=javascript#channel-capabilities) are included in `ChannelCapabilities` object. **NOTE:** _send-typing-events_ is deprecated and replaced with _typing-events_
4. [Optional] Experiment with Dashoboard's settings and check if we are receiving appropriate capabilities.

### ☑️Contributor Checklist

#### General

#### Code & documentation
- [x] New code is covered by unit tests (only parsing tests)

### ☑️Reviewer Checklist

### 🎉 GIF

_Please provide a suitable gif that describes your work on this pull request_
